File: Atividades/Lista2/Java/TestaCirculo.java
package Atividades.Lista2.Java;

import java.util.Scanner;

public class TestaCirculo {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        Circulo[] vetor1 = new Circulo[3];
        for (int i = 0; i < 3; i++) {
            int x = scanner.nextInt();
            int y = scanner.nextInt();
            int raio = scanner.nextInt();
            vetor1[i] = new Circulo(x, y, raio);
        }

        Circulo[] vetor2 = vetor1;

        Circulo[] vetor3 = new Circulo[3];
        for (int i = 0; i < 3; i++) {
            vetor3[i] = new Circulo(vetor1[i].getX(), vetor1[i].getY(), vetor1[i].getRaio());
        }

        for (int i = 0; i < 3; i++) {
            int novoRaio = scanner.nextInt();
            vetor1[i].setRaio(novoRaio);
        }

        imprimirVetor("vetor1", vetor1);
        imprimirVetor("vetor2", vetor2);
        imprimirVetor("vetor3", vetor3);

        scanner.close();
    }

    private static void imprimirVetor(String nome, Circulo[] vetor) {
        System.out.println(nome + ":");
        for (Circulo circulo : vetor) {
            System.out.println(circulo.getX() + " " + circulo.getY() + " " + circulo.getRaio());
        }
    }
}

class Circulo {
    private int x, y, raio;
    public static final double PI = 3.14159;

    public Circulo(int x, int y, int raio) {
        this.x = x;
        this.y = y;
        this.raio = raio;
    }

    public int getX() {
        return x;
    }

    public int getY() {
        return y;
    }

    public int getRaio() {
        return raio;
    }

    public void setRaio(int raio) {
        this.raio = raio;
    }

    public double circunferencia() {
        return 2 * PI * raio;
    }
}

File: Atividades/Lista_2/C++/testaCirculo.cpp
#include <iostream>
#include <vector>

using namespace std;

int i = 1;

class Circulo {
private:
    int x, y, raio;
public:
    static constexpr double PI = 3.14159;
    Circulo() {
        x = 0;
        y = 0;
        raio = 0;
    }
    Circulo(int x, int y, int raio) {
        this->x = x;
        this->y = y;
        this->raio = raio;
    }

    int getX() {
        return x;
    }

    int getY() {
        return y;
    }

    int getRaio() {
        return raio;
    }

    void setRaio(int raio) {
        this->raio = raio;
    }

    double cincunferencia() {
        return 2 * PI * raio;
    }
};

void imprimirVetor(vector<Circulo>& vetor) {
    cout << "vetor" << i << ":\n";
    for (Circulo& circulo : vetor) {
        cout << circulo.getX() << " " << circulo.getY() << " " << circulo.getRaio() << endl;
    }
    i++;
}

int main() {
    vector<Circulo> vetor1(3);

    for (int i = 0; i < 3; ++i) {
        int x, y, raio;
        cin >> x >> y >> raio;
        Circulo tempCirculo(x, y, raio);
        vetor1[i] = tempCirculo;
    }
    vector<Circulo>& vetor2 = vetor1;

    vector<Circulo> vetor3(3);
    for (int i = 0; i < 3; ++i) {
        vetor3[i] = vetor1[i];
    }

    for (int i = 0; i < 3; ++i) {
        int novoRaio;
        cin >> novoRaio;
        vetor1[i].setRaio(novoRaio);
    }

    imprimirVetor(vetor1);
    imprimirVetor(vetor2);
    imprimirVetor(vetor3);
}